src/graphorm/graphorm.ts

```diff
@@ -95,9 +96,21 @@ export class GraphORM {
         childColumn: fk.columnNames[0],
       };
     }
-    const inverseRelation = parentMetadata.relations.find(
+    const inverseRelations = parentMetadata.relations.filter(
       (rel) => rel.inverseEntityMetadata.name === childMetadata.name,
     );
+    const inverseRelation =
+      inverseRelations.length === 1
+        ? inverseRelations[0]
+        : inverseRelations.find((rel) => {
+            const fk = rel.foreignKeys[0].columnNames[0];
+            // the issue came from when we reference a table twice in the same entity (ex. Author and Source) - always using the first element of the array won't work
+            // strip the id part to compare the reference as it is a common convention on ORMs (ex. on EF Core ORM)
+            // ex. when we're looking for the "author" - it is highliy likely the relevant FK is named "authorId" - so we remove the "Id" part for comparison
+            // else we will have to force utilizing the metadata that would come from the "JoinColumn" provided by typeorm but have to explicitly define it
+            return fk.substring(0, fk.length - 2) === field;
```

Comment:
Nice catch! I think that at some point we will have to optimize graphorm. The runtime logic is growing and basically many of it can be done on boot